### PR TITLE
WT-10139 Add record count field for table

### PR DESCRIPTION
**Pull request by mm-ng**

### CODE REVIEW

File: src/include/btmem.h [hunk 1]
```diff
@@ -124,11 +125,23 @@ __wt_page_header_byteswap(WT_PAGE_HEADER *dsk)
...
+    int64_t byte_count;
+    int64_t row_count;
+};
+
 /*
  * WT_ADDR --
  *	An in-memory structure to hold a block's location.
  */
 struct __wt_addr {
+    WT_PAGE_STAT ps; /* Page information including row and byte counts */
```
Comment (quchenhao) [inline src/include/btmem.h:143]: Why we put it at the start of the struct? I think we should move it to the end to ease backward compatibility.

File: src/include/btree_inline.h [hunk 2]
```diff
@@ -1490,6 +1490,8 @@ struct __wt_addr_copy {
     uint8_t addr[WT_BTREE_MAX_ADDR_COOKIE];
     uint8_t size;
 
+    WT_PAGE_STAT ps; /* Page information including row and byte counts */
```
Comment (quchenhao) [inline src/include/btree_inline.h:1493]: Same here

File: src/include/cell.h [hunk 3]
```diff
@@ -200,6 +201,8 @@ struct __wt_cell_unpack_common {
 struct __wt_cell_unpack_addr {
     WT_CELL_COMMON_FIELDS;
 
+    WT_PAGE_STAT ps; /* Page information including row and byte counts */
```
Comment (quchenhao) [inline src/include/cell.h:204]: Same here

File: src/include/btmem.h (hunk 1 above)
Comment (mm-ng) [reply src/include/btmem.h]: Thanks for the suggestion, fixed it in a more recent commit.

File: src/include/cell_inline.h [hunk 4]
```diff
@@ -225,6 +225,16 @@ __wt_cell_pack_addr(WT_SESSION_IMPL *session, WT_CELL *cell, u_int cell_type, ui
         WT_IGNORE_RET(__wt_vpack_uint(&p, 0, recno));
     }
 
+    /*
```
Comment (quchenhao) [inline src/include/cell_inline.h:228]: Single row comment?

**Review by quchenhao — COMMENTED** [review]
Are we going to make the change to dump the stats in debug code in this pr?

**Comment by mm-ng** [comment]
> Are we going to make the change to dump the stats in debug code in this pr?

I think this was intended to be a pretty minimal implementation, my preference would be to tackle this in another ticket unless we feel strongly against this.

**Review by quchenhao — APPROVED** [review]
lgtm

**Review by quchenhao — COMMENTED** [review]
Where is the code that initialize the counters to -1?

**Comment by quchenhao** [comment]
> Where is the code that initialize the counters to -1?

It's OK. I found it.

File: src/include/cell_inline.h [hunk 5]
```diff
@@ -696,6 +705,8 @@ __wt_cell_unpack_safe(WT_SESSION_IMPL *session, const WT_PAGE_HEADER *dsk, WT_CE
     copy.len = 0; /* [-Wconditional-uninitialized] */
     copy.v = 0;   /* [-Wconditional-uninitialized] */
 
+    ps = unpack_addr == NULL ? NULL : &unpack_addr->ps;
```
Comment (quchenhao) [inline src/include/cell_inline.h:708]: We can move this to the else section in the following code. ps is not used in if (unpack_addr == NULL) case.

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Review by bvpvamsikrishna — COMMENTED** [review]
lgtm

**Review by GWlodarek — APPROVED** [review]

File: src/include/cell_inline.h (hunk 5 above)
Comment (mm-ng) [reply src/include/cell_inline.h:708]: Yeah I get what you mean but GCC was raising some compiler warnings when I had it in this section only earlier.

**Comment by mm-ng** [comment]
evergreen merge

**Pull request merged**